src/main/java/com/redhat/devtools/intellij/tektoncd/actions/component/StartTaskDialogAction.java

```diff
@@ -0,0 +1,59 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc.
+ ******************************************************************************/
+package com.redhat.devtools.intellij.tektoncd.actions.component;
+
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.redhat.devtools.intellij.tektoncd.ui.component.StartTaskDialog;
+import com.redhat.devtools.intellij.common.utils.UIHelper;
+import com.redhat.devtools.intellij.tektoncd.actions.TektonAction;
+import com.redhat.devtools.intellij.tektoncd.tkn.Tkn;
+import com.redhat.devtools.intellij.tektoncd.tree.TaskNode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.swing.tree.TreePath;
+import java.io.IOException;
+import java.util.concurrent.CompletableFuture;
+
+public class StartTaskDialogAction extends TektonAction {
+    Logger logger = LoggerFactory.getLogger(StartTaskDialogAction.class);
+
+    public StartTaskDialogAction() { super(TaskNode.class); }
+
+    @Override
+    public void actionPerformed(AnActionEvent anActionEvent, TreePath path, Object selected, Tkn tkncli) {
+        String namespace = ((TaskNode)selected).getParent().getParent().toString();
```

Comment:
By definition any action is running in the UI thread so there is no need to the executeInUI. But the bad part in that when you create the dialog, you will call tkncli that may block the UI. So I would rewrite as:
ExecHelper.submit(() -> {
call tkncli to get info the the dialog
UIHelper.executeInUI() -> create dialog and retrieve result)
execute the run task (as we are in a separate thread)
}